File: it/structures/java/alltypes-default/test/Entry.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.nio.ByteBuffer;
import java.time.Instant;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public class Entry {
    @JsonProperty("boolean_type")
    final Optional<Boolean> booleanType;
    @JsonProperty("string_type")
    final Optional<String> stringType;
    @JsonProperty("datetime_type")
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    final Optional<Instant> datetimeType;
    @JsonProperty("unsigned_32")
    final Optional<Integer> unsigned32;
    @JsonProperty("unsigned_64")
    final Optional<Long> unsigned64;
    @JsonProperty("signed_32")
    final Optional<Integer> signed32;
    @JsonProperty("signed_64")
    final Optional<Long> signed64;
    @JsonProperty("float_type")
    final Optional<Float> floatType;
    @JsonProperty("double_type")
    final Optional<Double> doubleType;
    @JsonProperty("bytes_type")
    final Optional<ByteBuffer> bytesType;
    @JsonProperty("any_type")
    final Optional<Object> anyType;
    @JsonProperty("array_type")
    final Optional<List<Entry>> arrayType;
    @JsonProperty("array_of_array_type")
    final Optional<List<List<Entry>>> arrayOfArrayType;
    @JsonProperty("map_type")
    final Optional<Map<String, Entry>> mapType;

    @JsonCreator
    public Entry(
        @JsonProperty("boolean_type") Optional<Boolean> booleanType,
        @JsonProperty("string_type") Optional<String> stringType,
        @JsonProperty("datetime_type") Optional<Instant> datetimeType,
        @JsonProperty("unsigned_32") Optional<Integer> unsigned32,
        @JsonProperty("unsigned_64") Optional<Long> unsigned64,
        @JsonProperty("signed_32") Optional<Integer> signed32,
        @JsonProperty("signed_64") Optional<Long> signed64,
        @JsonProperty("float_type") Optional<Float> floatType,
        @JsonProperty("double_type") Optional<Double> doubleType,
        @JsonProperty("bytes_type") Optional<ByteBuffer> bytesType,
        @JsonProperty("any_type") Optional<Object> anyType,
        @JsonProperty("array_type") Optional<List<Entry>> arrayType,
        @JsonProperty("array_of_array_type") Optional<List<List<Entry>>> arrayOfArrayType,
        @JsonProperty("map_type") Optional<Map<String, Entry>> mapType
    ) {
        this.booleanType = booleanType;
        this.stringType = stringType;
        this.datetimeType = datetimeType;
        this.unsigned32 = unsigned32;
        this.unsigned64 = unsigned64;
        this.signed32 = signed32;
        this.signed64 = signed64;
        this.floatType = floatType;
        this.doubleType = doubleType;
        this.bytesType = bytesType;
        this.anyType = anyType;
        this.arrayType = arrayType;
        this.arrayOfArrayType = arrayOfArrayType;
        this.mapType = mapType;
    }

    @JsonProperty("boolean_type")
    public Optional<Boolean> getBooleanType() {
        return this.booleanType;
    }

    @JsonProperty("string_type")
    public Optional<String> getStringType() {
        return this.stringType;
    }

    @JsonProperty("datetime_type")
    public Optional<Instant> getDatetimeType() {
        return this.datetimeType;
    }

    @JsonProperty("unsigned_32")
    public Optional<Integer> getUnsigned32() {
        return this.unsigned32;
    }

    @JsonProperty("unsigned_64")
    public Optional<Long> getUnsigned64() {
        return this.unsigned64;
    }

    @JsonProperty("signed_32")
    public Optional<Integer> getSigned32() {
        return this.signed32;
    }

    @JsonProperty("signed_64")
    public Optional<Long> getSigned64() {
        return this.signed64;
    }

    @JsonProperty("float_type")
    public Optional<Float> getFloatType() {
        return this.floatType;
    }

    @JsonProperty("double_type")
    public Optional<Double> getDoubleType() {
        return this.doubleType;
    }

    @JsonProperty("bytes_type")
    public Optional<ByteBuffer> getBytesType() {
        return this.bytesType;
    }

    @JsonProperty("any_type")
    public Optional<Object> getAnyType() {
        return this.anyType;
    }

    @JsonProperty("array_type")
    public Optional<List<Entry>> getArrayType() {
        return this.arrayType;
    }

    @JsonProperty("array_of_array_type")
    public Optional<List<List<Entry>>> getArrayOfArrayType() {
        return this.arrayOfArrayType;
    }

    @JsonProperty("map_type")
    public Optional<Map<String, Entry>> getMapType() {
        return this.mapType;
    }

    @Override
    public String toString() {
        final StringBuilder b = new StringBuilder();

        b.append("Entry(");
        b.append("boolean_type=");
        b.append(this.booleanType.toString());
        b.append(", ");
        b.append("string_type=");
        b.append(this.stringType.toString());
        b.append(", ");
        b.append("datetime_type=");
        b.append(this.datetimeType.toString());
        b.append(", ");
        b.append("unsigned_32=");
        b.append(this.unsigned32.toString());
        b.append(", ");
        b.append("unsigned_64=");
        b.append(this.unsigned64.toString());
        b.append(", ");
        b.append("signed_32=");
        b.append(this.signed32.toString());
        b.append(", ");
        b.append("signed_64=");
        b.append(this.signed64.toString());
        b.append(", ");
        b.append("float_type=");
        b.append(this.floatType.toString());
        b.append(", ");
        b.append("double_type=");
        b.append(this.doubleType.toString());
        b.append(", ");
        b.append("bytes_type=");
        b.append(this.bytesType.toString());
        b.append(", ");
        b.append("any_type=");
        b.append(this.anyType.toString());
        b.append(", ");
        b.append("array_type=");
        b.append(this.arrayType.toString());
        b.append(", ");
        b.append("array_of_array_type=");
        b.append(this.arrayOfArrayType.toString());
        b.append(", ");
        b.append("map_type=");
        b.append(this.mapType.toString());
        b.append(")");

        return b.toString();
    }

    @Override
    public int hashCode() {
        int result = 1;
        final StringBuilder b = new StringBuilder();
        result = result * 31 + this.booleanType.hashCode();
        result = result * 31 + this.stringType.hashCode();
        result = result * 31 + this.datetimeType.hashCode();
        result = result * 31 + this.unsigned32.hashCode();
        result = result * 31 + this.unsigned64.hashCode();
        result = result * 31 + this.signed32.hashCode();
        result = result * 31 + this.signed64.hashCode();
        result = result * 31 + this.floatType.hashCode();
        result = result * 31 + this.doubleType.hashCode();
        result = result * 31 + this.bytesType.hashCode();
        result = result * 31 + this.anyType.hashCode();
        result = result * 31 + this.arrayType.hashCode();
        result = result * 31 + this.arrayOfArrayType.hashCode();
        result = result * 31 + this.mapType.hashCode();
        return result;
    }

    @Override
    public boolean equals(final Object other_) {
        if (other_ == null) {
            return false;
        }

        if (!(other_ instanceof Entry)) {
            return false;
        }

        @SuppressWarnings("unchecked")
        final Entry o_ = (Entry)other_;

        if (!this.booleanType.equals(o_.booleanType)) {
            return false;
        }

        if (!this.stringType.equals(o_.stringType)) {
            return false;
        }

        if (!this.datetimeType.equals(o_.datetimeType)) {
            return false;
        }

        if (!this.unsigned32.equals(o_.unsigned32)) {
            return false;
        }

        if (!this.unsigned64.equals(o_.unsigned64)) {
            return false;
        }

        if (!this.signed32.equals(o_.signed32)) {
            return false;
        }

        if (!this.signed64.equals(o_.signed64)) {
            return false;
        }

        if (!this.floatType.equals(o_.floatType)) {
            return false;
        }

        if (!this.doubleType.equals(o_.doubleType)) {
            return false;
        }

        if (!this.bytesType.equals(o_.bytesType)) {
            return false;
        }

        if (!this.anyType.equals(o_.anyType)) {
            return false;
        }

        if (!this.arrayType.equals(o_.arrayType)) {
            return false;
        }

        if (!this.arrayOfArrayType.equals(o_.arrayOfArrayType)) {
            return false;
        }

        if (!this.mapType.equals(o_.mapType)) {
            return false;
        }

        return true;
    }

    public static class Builder {
        private Optional<Boolean> booleanType;
        private Optional<String> stringType;
        private Optional<Instant> datetimeType;
        private Optional<Integer> unsigned32;
        private Optional<Long> unsigned64;
        private Optional<Integer> signed32;
        private Optional<Long> signed64;
        private Optional<Float> floatType;
        private Optional<Double> doubleType;
        private Optional<ByteBuffer> bytesType;
        private Optional<Object> anyType;
        private Optional<List<Entry>> arrayType;
        private Optional<List<List<Entry>>> arrayOfArrayType;
        private Optional<Map<String, Entry>> mapType;

        private Builder() {
            this.booleanType = Optional.empty();
            this.stringType = Optional.empty();
            this.datetimeType = Optional.empty();
            this.unsigned32 = Optional.empty();
            this.unsigned64 = Optional.empty();
            this.signed32 = Optional.empty();
            this.signed64 = Optional.empty();
            this.floatType = Optional.empty();
            this.doubleType = Optional.empty();
            this.bytesType = Optional.empty();
            this.anyType = Optional.empty();
            this.arrayType = Optional.empty();
            this.arrayOfArrayType = Optional.empty();
            this.mapType = Optional.empty();
        }

        public Entry build() {

            return new Entry(
                this.booleanType,
                this.stringType,
                this.datetimeType,
                this.unsigned32,
                this.unsigned64,
                this.signed32,
                this.signed64,
                this.floatType,
                this.doubleType,
                this.bytesType,
                this.anyType,
                this.arrayType,
                this.arrayOfArrayType,
                this.mapType
            );
        }

        public Builder booleanType(final boolean booleanType) {
            this.booleanType = Optional.of(booleanType);
            return this;
        }

        public Builder stringType(final String stringType) {
            this.stringType = Optional.of(stringType);
            return this;
        }

        public Builder datetimeType(final Instant datetimeType) {
            this.datetimeType = Optional.of(datetimeType);
            return this;
        }

        public Builder unsigned32(final int unsigned32) {
            this.unsigned32 = Optional.of(unsigned32);
            return this;
        }

        public Builder unsigned64(final long unsigned64) {
            this.unsigned64 = Optional.of(unsigned64);
            return this;
        }

        public Builder signed32(final int signed32) {
            this.signed32 = Optional.of(signed32);
            return this;
        }

        public Builder signed64(final long signed64) {
            this.signed64 = Optional.of(signed64);
            return this;
        }

        public Builder floatType(final float floatType) {
            this.floatType = Optional.of(floatType);
            return this;
        }

        public Builder doubleType(final double doubleType) {
            this.doubleType = Optional.of(doubleType);
            return this;
        }

        public Builder bytesType(final ByteBuffer bytesType) {
            this.bytesType = Optional.of(bytesType);
            return this;
        }

        public Builder anyType(final Object anyType) {
            this.anyType = Optional.of(anyType);
            return this;
        }

        public Builder arrayType(final List<Entry> arrayType) {
            this.arrayType = Optional.of(arrayType);
            return this;
        }

        public Builder arrayOfArrayType(final List<List<Entry>> arrayOfArrayType) {
            this.arrayOfArrayType = Optional.of(arrayOfArrayType);
            return this;
        }

        public Builder mapType(final Map<String, Entry> mapType) {
            this.mapType = Optional.of(mapType);
            return this;
        }
    }

    /**
     * Construct a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }
}

File: it/structures/csharp/alltypes-default/Test/Entry.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace Test {
    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
    public class Entry {
        [JsonProperty("boolean_type")]
        public bool? booleanType {
            get;
        }

        [JsonProperty("string_type")]
        public String stringType {
            get;
        }

        [JsonProperty("datetime_type")]
        public DateTime? datetimeType {
            get;
        }

        [JsonProperty("unsigned_32")]
        public uint? unsigned32 {
            get;
        }

        [JsonProperty("unsigned_64")]
        public ulong? unsigned64 {
            get;
        }

        [JsonProperty("signed_32")]
        public int? signed32 {
            get;
        }

        [JsonProperty("signed_64")]
        public long? signed64 {
            get;
        }

        [JsonProperty("float_type")]
        public float? floatType {
            get;
        }

        [JsonProperty("double_type")]
        public double? doubleType {
            get;
        }

        [JsonProperty("bytes_type")]
        public byte[] bytesType {
            get;
        }

        [JsonProperty("any_type")]
        public Object anyType {
            get;
        }

        [JsonProperty("array_type")]
        public List<Entry> arrayType {
            get;
        }

        [JsonProperty("array_of_array_type")]
        public List<List<Entry>> arrayOfArrayType {
            get;
        }

        [JsonProperty("map_type")]
        public Dictionary<String, Entry> mapType {
            get;
        }

        [JsonConstructor]
        public Entry (
            [JsonProperty("boolean_type")] bool? booleanType,
            [JsonProperty("string_type")] String stringType,
            [JsonProperty("datetime_type")] DateTime? datetimeType,
            [JsonProperty("unsigned_32")] uint? unsigned32,
            [JsonProperty("unsigned_64")] ulong? unsigned64,
            [JsonProperty("signed_32")] int? signed32,
            [JsonProperty("signed_64")] long? signed64,
            [JsonProperty("float_type")] float? floatType,
            [JsonProperty("double_type")] double? doubleType,
            [JsonProperty("bytes_type")] byte[] bytesType,
            [JsonProperty("any_type")] Object anyType,
            [JsonProperty("array_type")] List<Entry> arrayType,
            [JsonProperty("array_of_array_type")] List<List<Entry>> arrayOfArrayType,
            [JsonProperty("map_type")] Dictionary<String, Entry> mapType
        ) {
            this.booleanType = booleanType;
            this.stringType = stringType;
            this.datetimeType = datetimeType;
            this.unsigned32 = unsigned32;
            this.unsigned64 = unsigned64;
            this.signed32 = signed32;
            this.signed64 = signed64;
            this.floatType = floatType;
            this.doubleType = doubleType;
            this.bytesType = bytesType;
            this.anyType = anyType;
            this.arrayType = arrayType;
            this.arrayOfArrayType = arrayOfArrayType;
            this.mapType = mapType;
        }

        public override bool Equals(Object other) {
            Entry o = other as Entry;

            if (o == null) {
                return false;
            }

            if (!this.booleanType.Equals(o.booleanType)) {
                return false;
            }

            if (this.stringType == null) {
                if (o.stringType != null) {
                    return false;
                }
            } else {
                if (!this.stringType.Equals(o.stringType)) {
                    return false;
                }
            }

            if (!this.datetimeType.Equals(o.datetimeType)) {
                return false;
            }

            if (!this.unsigned32.Equals(o.unsigned32)) {
                return false;
            }

            if (!this.unsigned64.Equals(o.unsigned64)) {
                return false;
            }

            if (!this.signed32.Equals(o.signed32)) {
                return false;
            }

            if (!this.signed64.Equals(o.signed64)) {
                return false;
            }

            if (!this.floatType.Equals(o.floatType)) {
                return false;
            }

            if (!this.doubleType.Equals(o.doubleType)) {
                return false;
            }

            if (this.bytesType == null) {
                if (o.bytesType != null) {
                    return false;
                }
            } else {
                if (!this.bytesType.Equals(o.bytesType)) {
                    return false;
                }
            }

            if (this.anyType == null) {
                if (o.anyType != null) {
                    return false;
                }
            } else {
                if (!this.anyType.Equals(o.anyType)) {
                    return false;
                }
            }

            if (this.arrayType == null) {
                if (o.arrayType != null) {
                    return false;
                }
            } else {
                if (!this.arrayType.Equals(o.arrayType)) {
                    return false;
                }
            }

            if (this.arrayOfArrayType == null) {
                if (o.arrayOfArrayType != null) {
                    return false;
                }
            } else {
                if (!this.arrayOfArrayType.Equals(o.arrayOfArrayType)) {
                    return false;
                }
            }

            if (this.mapType == null) {
                if (o.mapType != null) {
                    return false;
                }
            } else {
                if (!this.mapType.Equals(o.mapType)) {
                    return false;
                }
            }

            return true;
        }

        public override int GetHashCode() {
            int result = 1;
            result = result * 31 + this.booleanType.GetHashCode();
            result = result * 31 + this.stringType.GetHashCode();
            result = result * 31 + this.datetimeType.GetHashCode();
            result = result * 31 + this.unsigned32.GetHashCode();
            result = result * 31 + this.unsigned64.GetHashCode();
            result = result * 31 + this.signed32.GetHashCode();
            result = result * 31 + this.signed64.GetHashCode();
            result = result * 31 + this.floatType.GetHashCode();
            result = result * 31 + this.doubleType.GetHashCode();
            result = result * 31 + this.bytesType.GetHashCode();
            result = result * 31 + this.anyType.GetHashCode();
            result = result * 31 + this.arrayType.GetHashCode();
            result = result * 31 + this.arrayOfArrayType.GetHashCode();
            result = result * 31 + this.mapType.GetHashCode();
            return result;
        }

        public override String ToString() {
            StringBuilder b = new StringBuilder();

            b.Append("Entry(");
            b.Append("boolean_type=");
            b.Append(this.booleanType);
            b.Append(", ");
            b.Append("string_type=");
            b.Append(this.stringType);
            b.Append(", ");
            b.Append("datetime_type=");
            b.Append(this.datetimeType);
            b.Append(", ");
            b.Append("unsigned_32=");
            b.Append(this.unsigned32);
            b.Append(", ");
            b.Append("unsigned_64=");
            b.Append(this.unsigned64);
            b.Append(", ");
            b.Append("signed_32=");
            b.Append(this.signed32);
            b.Append(", ");
            b.Append("signed_64=");
            b.Append(this.signed64);
            b.Append(", ");
            b.Append("float_type=");
            b.Append(this.floatType);
            b.Append(", ");
            b.Append("double_type=");
            b.Append(this.doubleType);
            b.Append(", ");
            b.Append("bytes_type=");
            b.Append(this.bytesType);
            b.Append(", ");
            b.Append("any_type=");
            b.Append(this.anyType);
            b.Append(", ");
            b.Append("array_type=");
            b.Append(this.arrayType);
            b.Append(", ");
            b.Append("array_of_array_type=");
            b.Append(this.arrayOfArrayType);
            b.Append(", ");
            b.Append("map_type=");
            b.Append(this.mapType);
            b.Append(")");

            return b.ToString();
        }
    }
}
